### PR TITLE
Generate Higher-Level Review PDF v3

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

- This is part 2 of 3, a follow-up to #11960 - it adds structure for the new higher-level review PDF, but does not make use of the PDF in any APIs yet. I had to remove the following from this PR to get it under the line limit:
  - tests for structure.rb (these were copied from v2 of the form - you can see them in my branch for part 3 [here](https://github.com/department-of-veterans-affairs/vets-api/blob/95f910d836d0604ce7951d87a467323629415543/modules/appeals_api/spec/services/appeals_api/pdf_construction/higher_level_review/v3/structure_spec.rb)
  - flipper flags and any changes to API controllers ([here in my other branch](https://github.com/department-of-veterans-affairs/vets-api/blob/95f910d836d0604ce7951d87a467323629415543/modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews_controller.rb#L38-L39)
...I'll reintroduce these removed changes in a part 3 PR after this one is merged.
- As a reminder - there's one change in this new version of the PDF:
  - SOC/SSOC checkbox is removed (because you cannot opt out going forward)
- Other updates I made in this PR:
  -  Previously in v2, if a domestic claimant phone number had an extension, it wouldn't be printed on the form. For v3, I used the international phone number field as a fallback for this case (same behavior as in the new v3 of the SC form I was just working on) for both the veteran and claimant phone fields (see the `expected_200996_extra.pdf` fixture for an example)

## Related issue(s)
- [API-22858](https://vajira.max.gov/browse/API-22858)

## Testing done

- Added test cases based on v2 of the form + new ones for new phone behavior

## What areas of the site does it impact?
Adds a new Flipper flag, `decision_review_higher_level_review_pdf_v3`, which will cause the Higher-Level Review API v0 and Decision Reviews API v2 to start using v3 of the Higher-Level Review form PDF when submitting a form. Behavior will not change unless the flag is flipped.

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by caseywilliams** [comment]
It looks like I'll need to remove some content here so stay under the line limit - I'll break some of the out for a part 3.

**Comment by caseywilliams** [comment]
Ok, I've reduced the size of these changes: this PR no longer adds the new PDF to the API - it just generates the PDF. I'll make the last few changes in a part 3 PR after this one - I've updated the description to show which parts I left out here.

**Review by kristen-brown — APPROVED** [review]
I left a few comments to ponder, but this PR looks good to me overall. I know that we've already discussed the duplication of a large amount of the V2 form code for V3. I like how you used shared examples in the tests to reduce some of that duplication.

File: modules/appeals_api/app/services/appeals_api/pdf_construction/higher_level_review/v3/form_fields.rb [hunk 1]
```diff
@@ -70,6 +70,88 @@ def contestable_issue_month_field(subform, index)
...
+                  height: 15
+                }
+              end,
+              disagreement_area_pg2: Structure::MAX_ISSUES_SECOND_PAGE.times.map do |i|
+                {
+                  at: [table_left_pg2, table_top_pg2 - (row_height * i) - 21],
+                  width: left_col_width,
+                  height: 15
+                }
+              end,
```
[the remark this answers is not in the document]
Comment (cilestin) [reply modules/appeals_api/app/services/appeals_api/pdf_construction/higher_level_review/v3/form_fields.rb:149]: I think it was just an accidental pattern that's been repeated. We can absolutely extract these to other private methods.

[the remark this answers is not in the document]
Comment (caseywilliams) [reply modules/appeals_api/app/services/appeals_api/pdf_construction/higher_level_review/v3/form_fields.rb]: I'm struggling to keep this under the line limit (and can't remove any more content without breaking the tests), but for now I moved the boxes used in the contestable issues table into their own method and left a comment explaining why they're complex - I can update this in my next PR, though

**Review by cilestin — APPROVED** [review]

**Pull request merged**